app/models/descriptor.py

```diff
@@ -0,0 +1,103 @@
+from flask import current_app
+from .. import db
+
+class OptionAssociation(db.Model):
+    __tablename__ = 'optionassociations'
+    resource_id = db.Column(db.Integer, db.ForeignKey('resources.id'), primary_key=True)
+    descriptor_id = db.Column(db.Integer, db.ForeignKey('descriptors.id'), primary_key=True)
+    option = db.Column(db.Integer)
+    resource = db.relationship('Resource', back_populates='option_descriptors')
+    descriptor = db.relationship('Descriptor', back_populates='option_resources')
+
+    def __repr__(self):
+        return '%s: %s' % (self.descriptor.name, self.descriptor.values[self.option])
+
+class TextAssociation(db.Model):
+    __tablename__ = 'textassociations'
+    resource_id = db.Column(db.Integer, db.ForeignKey('resources.id'), primary_key=True)
+    descriptor_id = db.Column(db.Integer, db.ForeignKey('descriptors.id'), primary_key=True)
+    text = db.Column(db.String(64))
+    resource = db.relationship('Resource', back_populates='text_descriptors')
+    descriptor = db.relationship('Descriptor', back_populates='text_resources')
+
+    def __repr__(self):
+        return '%s: %s' % (self.descriptor.name, self.text)
+
+class Descriptor(db.Model):
+    __tablename__ = 'descriptors'
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(64), index=True)
+    values = db.Column(db.PickleType)
+    text_resources = db.relationship('TextAssociation', back_populates='descriptor')
+    option_resources = db.relationship('OptionAssociation', back_populates='descriptor')
+
+    def __repr__(self):
+        return '<Descriptor \'%s\'>' % self.name
+
+class Resource(db.Model):
+    __tablename__ = 'resources'
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(64), index=True)
+    address = db.Column(db.String(64))
+    lat = db.Column(db.Float)
+    long = db.Column(db.Float)
```

Comment:
Looks like `long` is overloaded here (since `long` is also a data type). Did PyCharm have any complaints about this? We could lengthen to `latitude` and `longitude` to avoid overloading issues.